public/assets/ts/editor.ts

```diff
@@ -1,21 +1,53 @@
 class Editor {
-  JSONEditor: any;
-  container: HTMLElement;
   options: any;
   editor: any;
+  sourceDataURL: string;
 
-  constructor() {
-    this.JSONEditor = window["JSONEditor"];
-    this.container = document.getElementById("jsoneditor") as HTMLElement;
+  constructor(
+    container: HTMLElement,
+    sourceDataURL: string,
+    initialJSON: string
+  ) {
     this.options = {};
-    this.editor = new this.JSONEditor(this.container, this.options);
+    this.sourceDataURL = sourceDataURL;
+    this.editor = new window["JSONEditor"](container, this.options);
+    this.editor.set(JSON.parse(initialJSON));
+    this.createSaveButton(container);
   }
 
-  useEditor(data): void {
-    this.editor.set(data);
+  private createSaveButton(container: HTMLElement) {
+    const saveButton = document.createElement("button");
+    saveButton.classList.add("save", "pink-button");
```

Comment:
this code is good for now, but I might remove it when making the editor.pug and put the button there - seems unnecessary to inject a button with js if it could just be in a pug file

Reply:
this doesn't require a change btw, feel free to resolve